examples/src/main/scala/org/apache/spark/examples/mllib/MovieLensALS.scala

```diff
@@ -171,18 +175,62 @@ object MovieLensALS {
 
     println(s"Test RMSE = $rmse.")
 
+    if (params.validateRecommendation) {
+      val (map, users) = computeRankingMetrics(model,
+        training, test, numMovies.toInt)
+      println(s"Test users $users MAP $map")
+    }
+    
     sc.stop()
   }
 
   /** Compute RMSE (Root Mean Squared Error). */
   def computeRmse(model: MatrixFactorizationModel, data: RDD[Rating], implicitPrefs: Boolean) = {
-
-    def mapPredictedRating(r: Double) = if (implicitPrefs) math.max(math.min(r, 1.0), 0.0) else r
-
     val predictions: RDD[Rating] = model.predict(data.map(x => (x.user, x.product)))
-    val predictionsAndRatings = predictions.map{ x =>
-      ((x.user, x.product), mapPredictedRating(x.rating))
+    val predictionsAndRatings = predictions.map { x =>
+      ((x.user, x.product), mapPredictedRating(x.rating, implicitPrefs))
     }.join(data.map(x => ((x.user, x.product), x.rating))).values
     math.sqrt(predictionsAndRatings.map(x => (x._1 - x._2) * (x._1 - x._2)).mean())
   }
+
+  def mapPredictedRating(r: Double, implicitPrefs: Boolean) = {
+    if (implicitPrefs) math.max(math.min(r, 1.0), 0.0)
```

Comment:
Fix style. Use `if .. else ..` without `{ .. }` only if it can fit into a single line. (https://cwiki.apache.org/confluence/display/SPARK/Spark+Code+Style+Guide#SparkCodeStyleGuide-CurlyBraces)